android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -142,4 +236,449 @@ public void rejectEvent(String eventId) {
+
+        Radar.RadarSearchPointsCallback callback = new Radar.RadarSearchPointsCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus status, @Nullable Location location, @Nullable RadarPoint[] points) {
+                if (status == Radar.RadarStatus.SUCCESS) {
+                    try {
+                        WritableMap map = Arguments.createMap();
+                        map.putString("status", status.toString());
+                        if (location != null) {
+                            map.putMap("location", RNRadarUtils.mapForJson(Radar.jsonForLocation(location)));
+                        }
+                        if (points != null) {
+                            map.putArray("points", RNRadarUtils.arrayForJson(RadarPoint.toJson(points)));
+                        }
+                        promise.resolve(map);
+                    } catch (JSONException e) {
+                        promise.reject(Radar.RadarStatus.ERROR_SERVER.toString(), Radar.RadarStatus.ERROR_SERVER.toString());
+                    }
+                } else {
+                    promise.reject(status.toString(), status.toString());
+                }
+            }
+        };
+
+        if (near != null) {
+            Radar.searchPoints(near, radius, tags, limit, callback);
+        } else {
+            Radar.searchPoints(radius, tags, limit, callback);
+        }
+    }
+
+    @ReactMethod
+    public void autocomplete(ReadableMap optionsMap, final Promise promise) {
+        if (promise == null) {
+            return;
+        }
+
+
+        if (!optionsMap.hasKey("query") || !optionsMap.hasKey("near")) {
+            promise.reject(Radar.RadarStatus.ERROR_BAD_REQUEST.toString(), Radar.RadarStatus.ERROR_BAD_REQUEST.toString());
+
+            return;
+        }
+
+        String query = optionsMap.getString("query");
+        ReadableMap nearMap = optionsMap.getMap("near");
+        double latitude = nearMap.getDouble("latitude");
+        double longitude = nearMap.getDouble("longitude");
+        Location near = new Location("RNRadarModule");
+        near.setLatitude(latitude);
+        near.setLongitude(longitude);
+        int limit = optionsMap.hasKey("limit") ? optionsMap.getInt("limit") : 10;
+
+        Radar.autocomplete(query, near, limit, new Radar.RadarGeocodeCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus status, @Nullable RadarAddress[] addresses) {
+                if (status == Radar.RadarStatus.SUCCESS) {
+                    try {
+                        WritableMap map = Arguments.createMap();
+                        map.putString("status", status.toString());
+                        if (addresses != null) {
+                            map.putArray("addresses", RNRadarUtils.arrayForJson(RadarAddress.toJson(addresses)));
+                        }
+                        promise.resolve(map);
```

Comment:
do we want to resolve if `addresses == null`? that seems like an error on our side?

Reply:
If status is success, should never be the case. Mainly just guarding for any downstream crashes